modules/grandpa/src/lib.rs

```diff
@@ -464,6 +467,20 @@ pub mod pallet {
 		})?)
 	}
 
+	/// Return number of headers to keep in the runtime storage.
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	pub(crate) fn headers_to_keep<T: Config<I>, I: 'static>() -> u32 {
+		T::HeadersToKeep::get()
+	}
+
+	/// Return number of headers to keep in the runtime storage.
+	#[cfg(feature = "runtime-benchmarks")]
+	pub(crate) fn headers_to_keep<T: Config<I>, I: 'static>() -> u32 {
+		// every db operation (significantly) slows down benchmarks, so let's keep as min as
+		// possible
+		2
+	}
```

Comment:
Nit: Could we do this distinction in the mock runtime instead ? Is the mock runtime used for benchmarks ?

Reply:
No, benchmarks are using the actual runtimes. That's why the weights of our pallets may be (are) different on Rialto, on Millau and on WBH/RBH. So every runtime generates its own `weights.rs`  file (e.g. for `frame-system` ; https://github.com/paritytech/polkadot/blob/master/runtime/rococo/src/weights/frame_system.rs, https://github.com/paritytech/substrate/blob/master/frame/system/src/weights.rs, https://github.com/paritytech/cumulus/blob/master/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/frame_system.rs)